### ISSUE TITLE
Cilium 1.14.9 failing `endpoint connectivity` everytime a pod is created 

### ISSUE THREAD

**Issue by primeroz** [cilium/cilium#31846]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

We recently upgraded our AWS clusters from cilium `1.11` to `1.14.9` and started to get alerts about `cilium endpoint unreachable` through the metric

```
# HELP cilium_unreachable_health_endpoints Number of health endpoints that cannot be reached
# TYPE cilium_unreachable_health_endpoints gauge
cilium_unreachable_health_endpoints 1
```

Doing more testing we were able to identify that this metric was going to > 0 every time  a pod is created on the node , looking at the details i can see that 

* Whenever a pod is created on dmesg we can see ( this is safe message since on 1.11 it happens as well but with no endpoint healthcheck problems ) 
```
[Mon Apr  8 16:48:29 2024] eth0: renamed from tmp70efd
[Mon Apr  8 16:48:29 2024] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Mon Apr  8 16:48:29 2024] IPv6: ADDRCONF(NETDEV_CHANGE): lxc3f309d46a56e: link becomes ready
```
* When this happens the `background cilium-health checks` to the node where the pod was created fails with `connection timed out` only to recover at the next `health-check run interval` in a minute or so 
```
  metric-1/ip-10-1-8-33.us-west-2.compute.internal:
    Host connectivity to 10.1.8.33:
      ICMP to stack:   Connection timed out
    Endpoint connectivity to 10.1.8.121:
      ICMP to stack:   Connection timed out
```
* during the `timeout period` a `ping from node to node` works just fine  and does not loose any ping nor see any increase in latency 
* a tcpdump shows that the `ICMP request and response` are being sent and received just fine with proper time gaps 

![image (2)](https://github.com/cilium/cilium/assets/2085772/c6e42dd6-6b01-4e70-87ed-04527232798e)

---

So _to me_ this connectivity check failure is a `red herring` and unrelated to any underlying network issue. 

Attached is a video to show the events

https://github.com/cilium/cilium/assets/2085772/50311d8d-97cc-48dc-a315-dd58dc76faf8

* around 00:44 we can see the endpoint connectivity fails 
  * even though there is a delay from `pod creation` to `connectivity endpoint failure` i _think_ that is because the background health check is only updated every 1 minute. 
  * the node is running fine for hours with no connectivity checks issue until a pod is created , at which point 100% the connectivity will fail at next update and recover at next one 
* It will recover around 1 minute later 

### Cilium Version

1.14.9 and 1.13.14 ( i will see if i can test 1.12 ) 

### Kernel Version

5.10.210-201.855.amzn2.x86_64 , 5.10.199-190.747.amzn2.x86_64 and 5.10.201-191.748.amzn2.x86_64

### Kubernetes Version

v1.26.12-eks-5e0fdde and v1.26.10-eks-e71965b with Control Plane at v1.26.13-eks-508b6b3

### Regression

1.11.14  ( probably whole 1.11.x but have not tested ) 

### Sysdump

It seems to collect **a lot of information** about cluste,r including network policy and such. 

I can't share those widely on a public github issue unfortunately ... 

### Relevant log output

```shell
debug log on the agent did not show anything useful to me
```

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by marseel** [comment]
FYI two related PRs that potentially fixed it in newer versions:
https://github.com/cilium/cilium/pull/30504
https://github.com/cilium/cilium/pull/30917

First one should change health status from unreachable to unknown instead and it's in the 1.15 release.
The second one wasn't released yet I think and will be part of 1.16

tldr core issue was that in eni mode whenever a new pod is scheduled, we update CiliumNode with pod IP, which was clearing health-checking data for that node.

**Comment by primeroz** [comment]
@marseel thank you 

Is it correct to say that this is red herring and i am safe to use 1.14.9 ( i will adjust the alert  ) ? 

thank you

**Comment by marseel** [comment]
Yes, seems like exactly the red herring that these two PRs were trying to fix. 

Do you have a possibility to test 1.15 or 1.16 prerelease?
In 1.15 you would see an unknown status reported instead, in 1.16 it should be working just fine.

**Comment by primeroz** [comment]
I will try 1.15 and report :+1:

**Comment by primeroz** [comment]
yes , in i1.15 i don't see the `connnection timeout` anymore, the metric is clean and i see

```
observability-1/ip-10-1-8-56.us-west-2.compute.internal                10.1.8.56         unknown     unknown
```

Thank you

**Comment by haribhusal2025** [comment]
as of v1.15.3, we are seeing same issue happening in our cluster.
